src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Table.cs

```diff
@@ -60,7 +60,7 @@ public override bool CheckTypes(TexlNode[] args, DType[] argTypes, IErrorContain
             for (var i = 0; i < argTypes.Length; i++)
             {
                 var argType = argTypes[i];
-                if (!argType.IsRecord)
+                if (!(argType.IsRecord || argType.IsDeferred))
```

Comment:
>|| argType.IsDeferred [](http://example.com/codeflow?start=39&length=21)

Do we need this?

Ie, why does Table() function need a special deferred check , but the other functions don't.

Reply:
Since Table() is the only function whose checkType is written in a way that it would not compute a correct return type if these changes are not made.